src/app/profile/settings/resources/resources.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, InjectionToken, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from 'ngx-login-client';
+import { Observable, Subscription } from 'rxjs';
+
+import { ResourceService } from '../services/resource.service';
+import { CpuStat } from './../../../space/create/deployments/models/cpu-stat';
+import { MemoryStat } from './../../../space/create/deployments/models/memory-stat';
+
+enum CLASSES {
+  ICON_OK = 'pficon-ok',
+  ICON_WARN = 'pficon-warning-triangle-o',
+  ICON_ERR = 'pficon-error-circle-o'
+}
+
+const DUMMY_UUID = 'f461b06c-3833-11e8-b467-0ed5f89f718b';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'resources-component',
+  templateUrl: 'resources.component.html',
+  styleUrls: ['./resources.component.less'],
+  providers: [
+    ResourceService
+  ]
+})
+export class ResourcesComponent implements OnInit, OnDestroy {
+
+  subscriptions: Subscription[] = [];
+
+  projectInfo: Object = {};
+  projectNames: string[] = [];
+
+  constructor(
+    private resourceService: ResourceService,
+    private userService: UserService
+  ) { }
+
+  ngOnInit() {
+    let loggedInUser = this.userService.currentLoggedInUser.attributes.username;
+
+    this.subscriptions.push(this.resourceService.getEnvironments().subscribe(
+      (projNames) => {
```

Comment:
I didn't look at this closely till now. This logic should be moved to the service. The view should just receive the object to be displayed.